app/models/middleware_performance.rb

```diff
@@ -19,7 +19,12 @@ def self.build_results_for_report_middleware(options)
     results = []
 
     model_class.find_each do |ms|
-      raw_stats = fetch_raw_stats(ms, start_time, end_time, interval)
+      begin
+        raw_stats = fetch_raw_stats(ms, start_time, end_time, interval)
+      rescue ::Hawkular::BaseClient::HawkularConnectionException => exception
+        $log.log_backtrace(exception)
```

Comment:
After checking, $log.error should be enough as i already have the backtrace on the exception.